### PR TITLE
 Fixed gen of traces with non-image vecenv (#913)

### LINKED ISSUE #913: [bug] Assertion error when generating expert trajectories in some vec envs
Hello, first of all, thanks for this great repo =)

**Describe the bug**
There is an assertion error (size of observation does not correspond to the size of actions) when trying to record expert trajectories in a vectorized environment where its observations are not images.

Inspecting the `generate_expert_traj` code, this assertion error occurs because when images are not recorded, [in this line](https://github.com/hill-a/stable-baselines/blob/b3b217c48752b50cca6a6eaac8264df5b8e88852/stable_baselines/gail/dataset/record_expert.py#L117), the observation for all the environments (and not only the first env as it does in line 109 with `obs_ = obs[0] if is_vec_env else obs`) is appended to the list. Therefore, when this happens and the observation array is reshaped in line 150, its len differs with the len of the actions.

**Code example**

```python
from stable_baselines.common.policies import MlpPolicy
from stable_baselines.common import make_vec_env
from stable_baselines.gail import generate_expert_traj
from stable_baselines import ACER

# multiprocess environment
env = make_vec_env('CartPole-v1', n_envs=4)

model = ACER(MlpPolicy, env, verbose=1)
model.learn(total_timesteps=25000)

generate_expert_traj(model, save_path='.', n_timesteps=0, n_episodes=5)

```

```bash
Traceback (most recent call last):
  File "/home/juan/.config/JetBrains/PyCharm2020.1/scratches/scratch_7.py", line 13, in <module>
    n_episodes=5)
  File "stable_baselines/gail/dataset/record_expert.py", line 163, in generate_expert_traj
    assert len(observations) == len(actions)
AssertionError

```

**System Info**
Describe the characteristic of your environment:
 * Describe how the library was installed: pip
 * GPU models and configuration: CPU (no GPU available)
 * Python version: 3.6.10
 * Tensorflow version: 1.15.2
 * Versions of any other relevant libraries: stable-baselines 2.10.0

### PR DESCRIPTION
**Pull request by jbarsce**
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) for new features and bug fixes)
Issue [#913 ](https://github.com/hill-a/stable-baselines/issues/913#issue-648209281)

closes #913 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the [changelog](https://github.com/hill-a/stable-baselines/blob/master/docs/misc/changelog.rst) accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have ensured `pytest` and `pytype` both pass (by running  `make pytest` and `make type`).

### CODE REVIEW

File: stable_baselines/gail/dataset/record_expert.py [hunk 1]
```diff
@@ -114,7 +114,8 @@ def generate_expert_traj(model, save_path=None, env=None, n_timesteps=0,
             cv2.imwrite(image_path, obs_)
             observations.append(image_path)
         else:
-            observations.append(obs)
+            obs_ = obs[0] if is_vec_env else obs
```
Comment (araffin) [inline stable_baselines/gail/dataset/record_expert.py:117]: you could put that statement before the if, no? (that would remove a duplication)

File: docs/misc/changelog.rst [hunk 2]
```diff
@@ -35,6 +35,7 @@ Bug Fixes:
 - Fixed ``seed()``` method for ``SubprocVecEnv``
 - Fixed a bug ``callback.locals`` did not have the correct values (@PartiallyTyped)
 - Fixed a bug in the ``close()`` method of ``SubprocVecEnv``, causing wrappers further down in the wrapper stack to not be closed. (@NeoExtended)
+- Fixed a bug in the ``generate_expert_traj()`` method in ``record_expert.py`` when using a non-image vectorized environment (@jbarsce)
```
Comment (araffin) [inline docs/misc/changelog.rst:38]: please add your username at the bottom too (unless it is already there)

File: stable_baselines/gail/dataset/record_expert.py (hunk 1 above)
Comment (jbarsce) [reply stable_baselines/gail/dataset/record_expert.py]: yes, you are right. Done

File: docs/misc/changelog.rst (hunk 2 above)
Comment (jbarsce) [reply docs/misc/changelog.rst:38]: done

**Comment by araffin** [comment]
Last thing, as written in the template, could you update the tests? Just add a small test recording a trajectory on `CartPole-v1` with a vec_env, it is in `tests/test_gail.py`. This tests should fail with master but pass with this PR ;)

**Comment by jbarsce** [comment]
> Last thing, as written in the template, could you update the tests? Just add a small test recording a trajectory on `CartPole-v1` with a vec_env, it is in `tests/test_gail.py`. This tests should fail with master but pass with this PR ;)

ready :)
if you need something else to be changed, please let me know (this is the first pull request I made on github :P)

**Pull request merged**